flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManagerDriver.java

```diff
@@ -689,4 +690,54 @@ public void onStopContainerError(ContainerId containerId, Throwable throwable) {
                     throwable);
         }
     }
+
+    public String getConatainerCompletedCause(ContainerStatus containerStatus) {
```

Comment:
```suggestion
    public String getContainerCompletedCause(ContainerStatus containerStatus) {
```

Reply:
Can't we make this function a static utility method? Then we could also test it quite easily in isolation and don't have to start the whole `YarnResourceManagerDriver`.